framework/src/executioners/FEProblemSolve.C

```diff
@@ -220,11 +221,14 @@ FEProblemSolve::FEProblemSolve(Executioner & ex)
 
   _problem.skipExceptionCheck(getParam<bool>("skip_exception_check"));
 
-  _problem.setMaxNLPingPong(getParam<unsigned int>("n_max_nonlinear_pingpong"));
-
-  _problem.setNonlinearForcedIterations(getParam<unsigned int>("nl_forced_its"));
-
-  _problem.setNonlinearAbsoluteDivergenceTolerance(getParam<Real>("nl_abs_div_tol"));
+  if (isParamValid("nonlinear_convergence"))
+  {
+    _problem.setNonlinearConvergenceName(getParam<ConvergenceName>("nonlinear_convergence"));
+    if (_problem.onlyAllowDefaultNonlinearConvergence())
+      mooseError("The selected problem does not allow 'nonlinear_convergence' to be set.");
+  }
```

Comment:
you also need to error on any parameter that got passed for the nonlinear solver in the Executioner but gets ignored when not using the DefaultNonlinearConvergence